python/python/Ice/Future.py

```diff
@@ -170,15 +335,15 @@ def add_done_callback(self, fn):
         Parameters
         ----------
         fn : callable
-            The function to execute upon completion.
+            The function to execute upon completion. Takes the future as its argument.
         """
         with self._condition:
             if self._state == Future.StateRunning:
                 self._doneCallbacks.append(fn)
                 return
         fn(self)
 
-    def result(self, timeout=None):
+    def result(self, timeout: int | float | None = None) -> _T:
```

Comment:
I don't understand why these are required here.

Reply:
why what are required?

Reply:
This end up in the wrong line, I think you updated the code while I was commenting.